### DISCUSSION TITLE
For posts without a featured image, is there a way to use the first image from the body as thumbnail?

### DISCUSSION THREAD

**Discussion by atkulp** [GetPublii/Publii#1727] [Help]
Like it says in the title. I have many posts without a featured image. I don't want to go back and remove an image so I can move it to the featured slot, but I want to see thumbnails in the grid/masonry view in themes like Mercury. Is that something a theme author needs to do, a general setting I'm missing, or impossible in current state?

**Comment by bjazmoore** [comment]
There is no built in function in Publii for doing this.  The featured image is the target image for thumbnails.  Can you do this with handlebars in the theme or in a child theme?  Maybe, but it would be a lot faster to set the featured image.  Sorry.

**Reply by akarshhemrajanime** [reply]
If there is only one image in a post, it should immediately be the featured image, so that there are no duplicates of the same image in one post. Hopefully there is a solution for this. Thank You

**Comment by kukkurovaca** [comment]
Yeah, this is a real PITA for themes that want to have a featured image. It's simply not always desirable for the first image to precede all the text.

I think one of these would be sensible:

- Per-post option to suppress the display of the featured image in the post itself. (There's already per-post override of the featured image width setting, we just need one that hides it.) Then it's not a problem if the featured image is a duplicate of one of the images in the post body. I assume this could be achieved relatively easily by modifying the theme? (For those who are up to doing that.)
- Treat first image as featured, as suggested above 
- Fallback to using the post title in place of the featured image

**Reply by akarshhemrajanime** [reply]
If there is only one image in a post, it should immediately be the featured image, so that there are no duplicates of the same image in one post. Hopefully there is a solution for this. Thank You
